posthog/models/user_home_settings.py

```diff
@@ -0,0 +1,27 @@
+from django.db import models
+
+from posthog.models.utils import RootTeamMixin, UUIDTModel
+
+
+class UserHomeSettings(UUIDTModel, RootTeamMixin):
+    """Stores personalized navigation settings such as pinned tabs and the homepage for a user within a team."""
+
+    user = models.ForeignKey(
+        "User",
+        on_delete=models.CASCADE,
+        related_name="home_settings",
+        null=True,
+        blank=True,
+    )
+    team = models.ForeignKey("Team", on_delete=models.CASCADE, null=True, blank=True)
+    tabs = models.JSONField(default=list, blank=True)
+    homepage = models.JSONField(default=dict, blank=True, null=True)
```

Comment:
From my experience, there's a bunch of stuff that one might throw into such fields. Doesn't it make sense to consolidate the two? Have one big JSON field with the user config? You could pick out whatever you need.

Reply:
Hmm... I'm so-so on this. I can see where this might be useful in the future, yet I also think there's nothing wrong with a little bit of structure. It makes it possible to only update certain fields without touching the rest of the JSON... and adds a bit of safety from the users... but not sure if that will be a game changer here. Unlikely. Adding new fields to the model when we need to isn't too bad, so I'm tempted to just leave it like it is, instead of going through another round of CI madness 😅

Reply:
I guess what I mean is that it feels cleaner to be able to do this:
<img width="281" height="122" alt="image" src="https://github.com/user-attachments/assets/7559ab67-6a55-49c1-8877-b9b63c249c3b" />

... and not carry around a big object in the frontend that we'll patch.